src/Angor/Client/Services/IMessageService.cs

```diff
@@ -0,0 +1,24 @@
+using Angor.Client.Models;
+using Nostr.Client.Messages;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Angor.Client.Services
+{
+    public interface IMessageService : IDisposable
+    {
+        IReadOnlyList<DirectMessage> DirectMessages { get; }
+        bool IsLoadingMessages { get; }
+        bool IsSendingMessage { get; }
+        bool IsRefreshing { get; }
+        bool IsSubscriptionActive { get; }
+        event Action OnChange;
```

Comment:
On change of what? can you add context to the name?

Reply:
renamed to `OnStateChange`